### PR TITLE
Change LaneCorridor selection in BaseIDM

### PR DESCRIPTION
**Pull request by patrickhart**
* if a static lane corridor is set, the LaneCorridor is now chosen using `ChooseLaneCorridorBasedOnVehicleState`
* ChooseLaneCorridorBasedOnVehicleState computes the lane corridor based on intersections and checks if any of the agent's polygon points in the front are in the other LaneCorridor

Observation: harsh braking does not occur directly at the beginning of the episode.

### CODE REVIEW

File: bark/world/map/commons.hpp [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ * @brief  Function that selects LaneCorridors
+ * @note   If an agent is in two LaneCorridors with its shape
+ *         the other one it selected if the front points of the
+ *         agent are within the other lane corridor.
+ *         This models an entering of the other lane corridor.
+ * @param  observed_world: ObservedWorld of agent
+ * @param  target_corr: fallback corridor in case calculation fails
+ * @retval LaneCorridorPtr: the computed lane corridor
+ */
+inline LaneCorridorPtr ChooseLaneCorridorBasedOnVehicleState(
```
Comment (klemense1) [inline bark/world/map/commons.hpp:42]: I like it! Do we have a unit test for it?

**Review by klemense1 — CHANGES_REQUESTED** [review]
Please consider writing a unit test for ChooseLaneCorridorBasedOnVehicleState. Also, did you check if BehaviorLaneChangeMobil is still working (in the examples)?

File: bark/world/map/commons.hpp (hunk 1 above)
Comment (patrickhart) [reply bark/world/map/commons.hpp:42]: I think we can discuss how we could unit test this next week.
Then it should be possible to still implement it until Thursday!

**Comment by patrickhart** [comment]
> Please consider writing a unit test for ChooseLaneCorridorBasedOnVehicleState. Also, did you check if BehaviorLaneChangeMobil is still working (in the examples)?

For now the selection is only active if a lane corridor has been set. So should not affect the behavior of BARK overall.
I think a system test validation is for now sufficient.

File: bark/world/map/commons.hpp (hunk 1 above)
Comment (patrickhart) [reply bark/world/map/commons.hpp:42]: No

**Comment by patrickhart** [comment]
Integrated in branch `rss_benchmark`.

**Pull request closed without merging**